packages/contentful-app-extensions/contributing-cohort-memberships/src/test/locations/Field.test.tsx

```diff
@@ -0,0 +1,266 @@
+import { FieldExtensionSDK, Entry } from '@contentful/app-sdk';
+import {
+  MultipleEntryReferenceEditor,
+  CustomEntityCardProps,
+  useEntity,
+} from '@contentful/field-editor-reference';
+import { useSDK, useAutoResizer } from '@contentful/react-apps-toolkit';
+
+jest.mock('@contentful/react-apps-toolkit', () => ({
+  useSDK: jest.fn(),
+  useAutoResizer: jest.fn(),
+}));
+
+jest.mock('@contentful/field-editor-reference', () => ({
+  MultipleEntryReferenceEditor: jest.fn(),
+  useEntity: jest.fn(),
+}));
+
+const mockBaseSdk = () => ({
+  window: {
+    startAutoResizer: jest.fn(),
+  },
+  space: {
+    unpublishEntry: jest.fn(),
+    deleteEntry: jest.fn(),
+  },
+});
+
+describe('Field component', () => {
+  let sdk: jest.Mocked<FieldExtensionSDK>;
+
+  beforeEach(() => {
+    sdk = mockBaseSdk() as unknown as jest.Mocked<FieldExtensionSDK>;
+    (useSDK as jest.Mock).mockReturnValue(sdk);
+    (MultipleEntryReferenceEditor as jest.Mock).mockImplementation(() => (
+      <p>MultipleEntryReferenceEditor</p>
+    ));
+  });
+
+  it('enables automatic resizing', async () => {
+    render(<Field />);
+    expect(useAutoResizer).toHaveBeenCalled();
+  });
+
+  it('passes a custom card renderer to <MultipleEntryReferenceEditor />', async () => {
+    render(<Field />);
+
+    expect(MultipleEntryReferenceEditor).toHaveBeenCalled();
+    expect(
+      (MultipleEntryReferenceEditor as jest.Mock).mock.lastCall[0]
+        .renderCustomCard,
+    ).toEqual(CustomCard);
+  });
+
+  describe('CustomCard component', () => {
+    beforeEach(() => {
+      (useEntity as jest.Mock).mockImplementation((type, id) => {
+        if (id === 'contributing-cohort-1') {
+          return {
+            data: {
+              fields: {
+                name: {
+                  'en-US': 'My Contributing Cohort',
+                },
+              },
+            },
+          };
+        }
+      });
+    });
+
+    it('loads contributing cohort name from related entity and renders contributing cohort name and role', async () => {
+      const props = {
+        entity: {
+          fields: {
+            role: {
+              'en-US': 'Lead',
+            },
+            contributingCohort: {
+              'en-US': {
+                sys: {
+                  id: 'contributing-cohort-1',
+                },
+              },
+            },
+          },
+          sys: {
+            type: 'Entry',
+            publishedVersion: 1,
+            version: 1,
+          },
+        },
+        onEdit: jest.fn(),
+        onRemove: jest.fn(),
+      } as unknown as CustomEntityCardProps;
+
+      render(<CustomCard {...props} />);
+
+      await waitFor(() => {
+        expect(useEntity).toHaveBeenCalledWith(
+          'Entry',
+          'contributing-cohort-1',
+        );
+        expect(screen.getByText('My Contributing Cohort')).toBeInTheDocument();
+        expect(screen.getByText('Lead')).toBeInTheDocument();
+      });
```

Comment:
Seems like we've got multiple assertions inside this waitFor. 

Can we have only one inside? But I think we can probably remove it and use findByText (prefer findByRole).